src/lava/magma/runtime/runtime.py

```diff
@@ -407,11 +415,11 @@ def stop(self):
 
     def join(self):
         """Join all ports and processes"""
-        for port in self.runtime_to_service:
```

Comment:
I see you have created a new list open_ports but have not eliminated the old lists (runtime_to_service and service_to_runtime). If there are no new items, and the two old containers are by convention, even if you delete the references from open ports, the references are still there in the old lists. So I don't get what the open ports is accomplishing here?

Reply:
* open_ports is only to track ports that need to be joined when the runtime is stopped. The join operation in the ports implements clean up logic needed to prevent leaking. This is not about deleting references to the ports, as they don't themselves implement resource clean up in their destructors (they aren't RAII objects).
* runtime_to_service and service_to_runtime looked like they were also intended to track ports to be joined, but they also serve other purposes.
* by introducing open_ports, I merely created a separate variable for the tracking of ports, while leaving the remaining accounting with runtime_to_service and service_to_runtime.
* The reason for splitting into a separate variable was that the the service_to_runtime and runtime_to_service were missing part of the ports to track.

Reply:
@ssgier Could you please address this question before we merge the PR?

Reply:
My message above was intended to address this. Is anything unclear?

Reply:
No sorry, I had apparently not reloaded the browser when I wrote my message and I had not seen your reply. My bad. Your answer sounds fine to me.

Reply:
@joyeshmishra Could you please review this and merge the PR if everything is in order?
Things look good to me.

Reply:
Looks good to me. Please go ahead.